pkg/microservice/systemconfig/core/features/service/feature.go

```diff
@@ -0,0 +1,105 @@
+package service
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/koderover/zadig/pkg/microservice/systemconfig/core/features/repository/models"
+	"github.com/koderover/zadig/pkg/microservice/systemconfig/core/features/repository/mongodb"
+	"github.com/koderover/zadig/pkg/tool/log"
+)
+
+type Feature string
+
+const (
+	ModernWorkflow             Feature = "ModernWorkflow"
+	CommunityProjectRepository Feature = "CommunityProjectRepository"
+	UserRegistration           Feature = "UserRegistration"
+)
+
+type FeatureGates map[Feature]bool
+
+var Features FeatureGates = map[Feature]bool{
+	ModernWorkflow:             false,
+	CommunityProjectRepository: false,
+	UserRegistration:           true,
+}
+
+func (fg FeatureGates) EnabledFeatures() []Feature {
+	var res []Feature
+
+	for k, v := range fg {
+		if v {
+			res = append(res, k)
+		}
+	}
+
+	return res
+}
+
+func (fg FeatureGates) FeatureEnabled(f Feature) bool {
+	return fg[f]
+}
+
+// MergeFeatureGates merge feature config from different source
+// latter feature configs will be overridden by former ones
+func (fg FeatureGates) MergeFeatureGates(fs ...FeatureGates) {
+	for _, v := range fs {
+		for k, vv := range v {
+			fg[k] = vv
+		}
+	}
+}
+
+func DBToFeatureGates() FeatureGates {
```

Comment:
If error may happen in the function, it's better to return error other than suppressing it.